crates/symbolicator-service/src/services/caches/sourcefiles.rs

```diff
@@ -42,22 +55,47 @@ impl SourceFilesCache {
         download_svc: Arc<DownloadService>,
     ) -> Self {
         Self {
-            cache: Arc::new(Cacher::new(cache, shared_cache)),
+            cache: Cacher::new(cache, shared_cache),
             download_svc,
         }
     }
 
     /// Retrieves the given [`RemoteFile`] from cache, or fetches it and persists it according
     /// to the provided [`Scope`].
     pub async fn fetch_file(&self, scope: &Scope, file: RemoteFile) -> CacheEntry<ByteViewString> {
+        // FIXME: We should probably make it possible to somehow predictably re-fetch
```

Comment:
> respect the caching headers from external servers

That'll be quite tricky tbh, as we always want the exact file at the time exception happened.

There can be a case that we'll let's say fetch and cache a `app.js` file that has 24h cache expiration header, but it has been already updated on the server. For all new unique users that trigger an exception, this would possibly be somehow out-of-sync and they frames would point to incorrect lines in the source code, but at the same time, I'm not sure how possible this scenario is.

I think for now it's fine as is and we may actually try to overengineer ourselves here 😅

Reply:
Even then, you might have users with an older version cached. Or you have some send delay for whatever reason, and you are not able to fetch the *older* file at all. So yeah, we can revisit this later when it turns out to be a real problem, otherwise lets not overengineer :-D